src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -82,12 +82,23 @@ internal static TextReader GetOrCreateReader()
             }
             else
             {
-                return StdInReader;
+                return TextReader.Synchronized(StdInReader);
```

Comment:
At this point, it looks like all outputs of GetOrCreateReader on both Unix and Windows are being wrapped in TextReader.Synchronized.  As such, we could refactor this so that GetOrCreateReader doesn't call TextReader.Synchronized, and that's instead called in the one place GetOrCreateReader is called, in Console.In:
https://github.com/dotnet/corefx/blob/0b868a47b4889fef013539787cf3475d817d893f/src/System.Console/src/System/Console.cs#L29